### DISCUSSION TITLE
"red" lights for input level missing in compact view

### DISCUSSION THREAD

**Discussion by wschell** [jamulussoftware/jamulus#1220] [UI Related]
In fancy view there are 5 green, 2 yellow and 2 red "lights" for input level metering for left and right input. 
When I change to compact view, the metering is always green. It would help a lot, if the color would change as in fancy view.
In the personal mix level meters it has at least 2 colors: green from bottom up and above read it to loud.
- Walter

**Comment by ann0see** [comment]
I'm not sure if this is possible easily since we're using parts of the normal system UI there.

**Reply by wschell** [reply]
The idea came, as I saw the level meter for one very loud person most green and red on top. Maybe it's a similar control for the input level.

**Reply by ann0see** [reply]
Yes. That's the clipping indicator. Maybe it's possible to implement this on the normal/compact UI too.

**Comment by JohannesBrx** [comment]
Theoretically it should be possible by modifying `CLevelMeter::SetBarMeterStyleAndClipStatus` and setting the appropriate style sheet attributes. Currently it is only implemented for the clipping indication in the normal and compact UI. However, I am not sure how computationally expensive it is to change the style sheet for all faders multiple times a second.

See https://github.com/jamulussoftware/jamulus/blob/master/src/levelmeter.cpp#L133.

**Reply by marcoboschi-1960** [reply]
johannesBrx , i don't want to waste your time , and of course first of all i want to thanks all of you for the great work and effort in jamulus...

i never  accessed the code , so maybe i 'm saying a completely wrong proposal  , but maybe , could be easier and better performing , splitting the single graphical level indicator object  in three different obj (one for green one for yellow and one for red) and setting them accordingly to input level  (just some aritmethic subtraction to set the correct value)... just like  ,i guess, you do in fancy view with the led ...

**Reply by JohannesBrx** [reply]
Unfortunately the three operating systems render the level indicators slightly different, I cannot imagine that you will achieve a convincing (and gap-less) appearance by stacking three individual indicators vertically.

**Reply by dcorson-ticino-com** [reply]
At present the level indicators for compact and normal skins only indicate OK and clipping with color.  The color change is done through a style sheet.  I think that the style sheets are interpreted at run time every time they are called so this would be much computation especially for the compact skin where we risk to have very many users.
There is certainly another way to do this, but someone must find it important enough to spend the time to do it.
IMHO this color change is merely a nice-to-have as the information is already given by the position of the level indicator.

**Reply by wschell** [reply]
Without changing color it would be helpful, to have something like colored "led" above or under the fader, which signals green, yellow and red and stays at yellow or red for a few seconds.
